clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1251,17 +1261,29 @@ class SyclKernelFieldChecker : public SyclKernelFieldHandler {
     assert(Util::isSyclAccessorType(Ty) &&
            "Should only be called on SYCL accessor types.");
 
-    const RecordDecl *RecD = Ty->getAsRecordDecl();
-    if (const ClassTemplateSpecializationDecl *CTSD =
-            dyn_cast<ClassTemplateSpecializationDecl>(RecD)) {
+    RecordDecl *RecD = Ty->getAsRecordDecl();
+    if (auto *CTSD = dyn_cast<ClassTemplateSpecializationDecl>(RecD)) {
       const TemplateArgumentList &TAL = CTSD->getTemplateArgs();
+
       TemplateArgument TA = TAL.get(0);
       const QualType TemplateArgTy = TA.getAsType();
-
-      if (TAL.size() > 5)
-        checkPropertyListType(TAL.get(5), Loc.getBegin());
       llvm::DenseSet<QualType> Visited;
       checkSYCLType(SemaRef, TemplateArgTy, Loc, Visited);
+
+      if (TAL.size() < 6) {
```

Comment:
Looks like we need to merge SYCL headers part first before enabling this diagnostics.

Reply:
I temporarily allowed old behavior, because to merge new headers we actually need this patch.